src/hooks/search/index.ts

```diff
@@ -130,13 +130,13 @@ const useSearch = (
                 const filterProduct = filterTypeItems.find(
                     l1 =>
                         l1.name === item ||
-                        l1.subItems.find(l2 => l2.name === item)
+                        l1.subFilters?.find(l2 => l2.name === item)
                 );
                 if (filterProduct) {
                     if (filterProduct.name !== item) {
                         // This means it's an L2 match.
                         filtersList.push(
-                            filterProduct.subItems.find(
+                            filterProduct.subFilters?.find(
```

Comment:
Guess it technically doesn't matter, but wouldn't you want to chain it on the first one because you might still run into the chance of `subFilters` being undefined, `filterProduct?.subFilters.find(...)`

Reply:
So if I'm [reading this right](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining), I think it should still be `subFilters?.` because the optional chain checks to see if the property with the `?` appended is undefined before trying to access a further property of it. `filterProduct` will always be defined, but `filterProduct.subFilters` might not be so we add the `?` to `subFilters`?

Reply:
oh word okay my mistake then, I always figured you would want to first check the property existing and if it exists, then you can safely call the method on it. Thanks for clarifying and ignore any other comments of mine lol

Reply:
lol all good, i feel like they could have done a better job making optional chaining more intuitive, I'm always looking at that page for reference.